pkg/kubelet/dockertools/manager.go

```diff
@@ -1769,6 +1756,73 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, podStatus *kub
 	}, nil
 }
 
+// tearDownPodNetwork calls the network plugin to tear down the pod (if
+// applicable) and delete the infra container.
+func (dm *DockerManager) tearDownPodNetwork(pod *api.Pod, runningPod *kubecontainer.Pod, networkSpec *api.Container, container *kubecontainer.Container) error {
+	ins, err := dm.client.InspectContainer(container.ID.ID)
+	if err != nil {
+		return fmt.Errorf("Error inspecting container %v: %v", container.ID.ID, err)
+	}
+	if getDockerNetworkMode(ins) != namespaceModeHost {
+		// pod could be nil, so we use the runningPod for name and namespace.
+		if err := dm.networkPlugin.TearDownPod(runningPod.Namespace, runningPod.Name, kubecontainer.DockerID(container.ID.ID)); err != nil {
+			return fmt.Errorf("Failed to teardown network for pod %q using network plugins %q: %v", pod.UID, dm.networkPlugin.Name(), err)
+		}
+	}
+	if err := dm.KillContainerInPod(container.ID, networkSpec, pod, "Need to kill pod."); err != nil {
+		return err
+	}
+	return nil
+}
+
+// setUpPodNetwork creates an infra container and calls the network plugin to
+// join the network if necessary. It returns the container ID of the infra
+// container and the pod IP.
+func (dm *DockerManager) setUpPodNetwork(pod *api.Pod) (kubecontainer.DockerID, string, error) {
+	glog.V(4).Infof("Creating pod infra container for %q", format.Pod(pod))
+	containerID, err, _ := dm.createPodInfraContainer(pod)
+	if err != nil {
+		glog.Errorf("Failed to create pod infra container: %v; Skipping pod %q", err, format.Pod(pod))
+		return containerID, "", err
```

Comment:
containerID should "" here but it would be clearer to just return nil/"" or whatever the NULL value for kubecontainer.DockerID is.